jx_sqlite/query_table.py

```diff
@@ -348,39 +348,46 @@ def query_metadata(self, query):
         frum, query['from'] = query['from'], self
         schema = self.sf.tables["."].schema
         query = QueryOp.wrap(query, schema)
-
-        if query.edges or query.groupby:
-            Log.error("Aggregates(groupby or edge) are not supported")
+        columns = self.sf.columns
         where = query.where
         tableName = None
         columnName = None
+
+        if query.edges or query.groupby:
+            Log.error("Aggregates(groupby or edge) are not supported")
+
         if where.op == "eq" and where.lhs.var == "table":
             tableName = mo_json.json2value(where.rhs.json)
         elif where.op =="eq" and where.lhs.var =="name":
             columnName = mo_json.json2value(where.rhs.json)
         else:
             Log.error("Only simple filters are expected like: \"eq\" on table and column name")
-        
+
+        t = [i for i in columns[0].names.keys()]
+        tables = [concat_field(self.sf.fact, i) for i in t]
+
         metadata = []
-        result = self.db.query("""SELECT NAME FROM SQLITE_MASTER WHERE type='table' ORDER BY NAME;""")
-        tables = wrap([{k: d[i] for i, k in enumerate(result.header)} for d in result.data])        
+        if columns[-1].es_column!="_id":
+            columns.append(Column(
+                names={i: relative_field("_id", i) for i in t},
+                type="string",
+                es_column="_id",
+                es_index=self.sf.fact,
+                nested_path=["."]
+            ))
+
         for table in tables:
-            if table.name.startswith("__") or (tableName != None and tableName != table.name):
+            if tableName != None and tableName != table:
                 continue
-            nested_path = [join_field(split_field(tab.name)[1:]) for tab in jx.reverse(tables) if startswith_field(table.name, tab.name)]
-            command = "PRAGMA table_info(" + quote_table(table.name) + ")"
-            columns = self.db.query(command)
-            for cid, name, dtype, notnull, dfft_value, pk in columns.data:
-                if name.startswith("__"):
-                    continue
-                cname, ctype = untyped_column(name)
+
+            tname=join_field(split_field(table)[1:])        
```

Comment:
instead of 

    tname=join_field(split_field(table)[1:]) 

use the `t` array directly: 

    for tname, table in zip(t, tables):